devito/types/grid.py

```diff
@@ -502,7 +502,6 @@ def __subdomain_finalize__(self, grid, **kwargs):
         # Create the SubDomain's SubDimensions
         sub_dimensions = []
         sdshape = []
-        counter = kwargs.get('counter', 0) - 1
```

Comment:
looks like remooving some hack here

Reply:
Yeah, that's the idea with the PR. It's also necessary to remove before introducing functions on subdimensions, since otherwise the counter gets even hackier